### DISCUSSION TITLE
Cookie mode is unclear

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#5328] [Archive]
I was trying to implement the cookie mode as described here: https://docs.directus.io/reference/api/system/authentication/

I was expecting the cookies to behave like a session but instead is only used for refresh the current token.

I think we could improve documentation or implement sessions with cookies. What are your thoughts?

[keywords: authentication, login, accountability, cookie, request, session, access]

**Comment by rijkvanzanten** [comment]
> I think we could improve documentation or implement sessions with cookies. What are your thoughts?

We can definitely improve the documentation 👍🏻 

Only the refresh token is handled in the cookie by design. This is to prevent access to the refresh token from JavaScript. That being said, you might need to access the actual access token from JS in order to do things like append tokens to image links for authentication (and other requests where cookies aren't available). 👍🏻

**Comment by odama626** [comment]
I recently ran into this as well.  I think having a cookie mode where the access_token is only accessible via an httpOnly cookie would be really useful too
